### PR TITLE
automation - add changelog and release automation

### PR DESCRIPTION
**Pull request by andrewkroh**
Add a new Github workflow for releases that automates the generation of a draft GH release with changelog notes.

Pull requests will be required to add new file to at `.changelog/<pr_number>.txt` with a changelog message or label with `skip-changelog` to bypass the check. See CONTRIBUTING.md for more details.

This backfills the .changelog directory with entries for changes since v0.14.0.

(I cloned this from go-sysinfo.)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Contributing
+
+Pull requests are welcomed. You must
+
+- Sign the Elastic [Contributor License Agreement](https://www.elastic.co/contributor-agreement).
+- Include a [changelog][changelog_docs] entry at `.changelog/{pr-number}.txt` with your pull request.
+- Include tests that demonstrate the change is working.
+
+[changelog_docs]: https://github.com/GoogleCloudPlatform/magic-modules/blob/2834761fec3acbf35cacbffe100530f82eada650/.ci/RELEASE_NOTES_GUIDE.md#expected-format
```
Comment (mjwolf) [inline CONTRIBUTING.md:9]: I don't see this line at all on the rendered github view: https://github.com/andrewkroh/stream/blob/release-workflow/CONTRIBUTING.md

But maybe there could be a template or examples of the format in this repo. I don't find the linked page very clear. Should there be a PR description before the release note? From the changelog in this PR, I guess  not, but the doc makes it seem like it should.

Comment (andrewkroh) [reply CONTRIBUTING.md]: I have added more explanation and examples in the CONTRIBUTING.md. After a few changes are made there will be many examples to follow in the .changelog/ dir.

**Review by efd6 — COMMENTED** [review]
It seems a shame to lose the old change log info (I know it's in the releases page — this is not committed).

**Review by mjwolf — APPROVED** [review]

**Comment by andrewkroh** [comment]
> It seems a shame to lose the old change log info (I know it's in the releases page — this is not committed).

I could move it to something like `.changelog/.pre-0.15.0-changelog.md`?

**Comment by efd6** [comment]
That SGTM

**Pull request merged**